src/tests/ftest/deployment/ior_per_rank.py

```diff
@@ -33,52 +33,46 @@ def execute_ior_per_rank(self, rank):
-                    actual_write_x = percent_change(dfs_max_write, self.expected_iops)
-                    actual_read_x = percent_change(dfs_max_read, self.expected_iops)
-
-                # compare actual and expected perf data
-                self.assertLessEqual(abs(actual_write_x), self.write_x,
-                                     "Max Write Diff too large for rank: {}".format(rank))
-                self.assertLessEqual(abs(actual_read_x), self.read_x,
-                                     "Max Read Diff too large for rank: {}".format(rank))
-                # collect list of good nodes
-                good_node = self.server_managers[0].get_host(rank)
-                if ((good_node not in self.good_nodes)
-                        and (good_node not in self.failed_nodes)):
-                    self.good_nodes.append(good_node)
-            except (TestFail, DaosTestError):
-                # collect bad nodes
-                failed_node = self.server_managers[0].get_host(rank)
-                if failed_node not in self.failed_nodes:
-                    self.failed_nodes[failed_node] = [rank]
-                else:
-                    self.failed_nodes[failed_node].append(rank)
-                if failed_node in self.good_nodes:
-                    self.good_nodes.remove(failed_node)
+        # execute ior on given rank and collect the results
+        try:
+            self.ior_cmd.flags.update(self.write_flags)
+            dfs_out = self.run_ior_with_pool(fail_on_warning=self.log.info)
+            dfs_perf_write = IorCommand.get_ior_metrics(dfs_out)
+            self.ior_cmd.flags.update(self.read_flags)
+            dfs_out = self.run_ior_with_pool(create_cont=False, fail_on_warning=self.log.info)
+            dfs_perf_read = IorCommand.get_ior_metrics(dfs_out)
+
+            # Destroy container, to be sure we use newly created container in next iteration
+            self.container.destroy()
+            self.container = None
+
+            # gather actual and expected perf data to be compared
+            dfs_max_write = float(dfs_perf_write[0][IorMetrics.MAX_MIB])
+            dfs_max_read = float(dfs_perf_read[0][IorMetrics.MAX_MIB])
+            actual_write_x = percent_change(dfs_max_write, self.expected_bw)
+            actual_read_x = percent_change(dfs_max_read, self.expected_bw)
+
+            # compare actual and expected perf data
+            self.assertLessEqual(
+                abs(actual_write_x), self.write_x,
+                "Max Write Diff too large for rank: {}".format(rank))
+            self.assertLessEqual(
+                abs(actual_read_x), self.read_x,
+                "Max Read Diff too large for rank: {}".format(rank))
+            # collect list of good nodes
+            good_node = self.server_managers[0].get_host(rank)
+            if ((good_node not in self.good_nodes)
+                    and (good_node not in self.failed_nodes)):
+                self.good_nodes.append(good_node)
```

Comment:
Can we print the failed nodes performance? The output has failed nodes information without the node performance information. Someone has to go over the job.log to find the write/read performance now.

Reply:
Yeah, I might as well while I'm here. Will push an update

Reply:
Added better reporting, but I'll need to verify it works